app/controllers/api/v1/results_controller.rb

```diff
@@ -139,8 +158,14 @@ def result_file_params
         prms = params[:included]&.select { |el| el[:type] == 'result_files' }&.first
         return nil unless prms
 
-        prms.require(:attributes).require(:file)
-        prms.dig(:attributes).permit(:file)
+        if prms.require(:attributes)[:file]
+          @form_multipart_upload = true
+          return prms.dig(:attributes).permit(:file)
+        end
+        attr_list = %i(file_data file_type file_name)
+
+        prms.require(:attributes).require(attr_list)
+        prms.dig(:attributes).permit(attr_list)
```

Comment:
attributes were already checked by require on previous line

Reply:
Yes, but we want to permit only what is in the attr_list? Or what do you mean? I am not sure. 

Another question. Do you think is ok to send `file_data`, `file_type`, and `file_name`. Or we want to combine data and type in URL base64 string? 
I didn't find what is "more correct" way, but I thin URL base64 string is more for inline images in HTML.

Reply:
I mean that you can simply call prms.permit(attributes: attr_list)

Reply:
Then we have to call `prms.permit(attributes: attr_list)[:attributes]` because we want to return only nested hash in attributes?